### PR TITLE
Updates for WC 3.0 compatibility.

### PR DESCRIPTION
**Pull request by bor0**

### CODE REVIEW

File: includes/class-wc-addons-gateway-simplify-commerce.php [hunk 1]
```diff
@@ -176,10 +179,13 @@ protected function save_subscription_meta( $order_id, $customer_id ) {
...
 	protected function process_pre_order( $order, $cart_token = '' ) {
-		if ( WC_Pre_Orders_Order::order_requires_payment_tokenization( $order->id ) ) {
+		$pre_wc_30 = version_compare( WC_VERSION, '3.0', '<' );
+		$order_id = $pre_wc_30 ? $order->id : $order->get_id();
+
+		if ( WC_Pre_Orders_Order::order_requires_payment_tokenization( $order_id ) ) {
 
 			try {
-				if ( $order->order_total * 100 < 50 ) {
+				if ( absint( $order->get_total() * 100 ) < 50 ) {
```
Comment (gedex) [inline includes/class-wc-addons-gateway-simplify-commerce.php]: don't use `absint`, `get_total` returns float. What if order total is `50.05`.

Comment (gedex) [inline includes/class-wc-addons-gateway-simplify-commerce.php]: it still false though 😄

File: includes/class-wc-addons-gateway-simplify-commerce.php [hunk 2]
```diff
@@ -362,7 +374,8 @@ public function scheduled_subscription_payment( $amount_to_charge, $renewal_orde
 	 * @param WC_Order $renewal_order The order which recorded the successful payment (to make up for the failed automatic payment).
 	 */
 	public function update_failing_payment_method( $subscription, $renewal_order ) {
-		update_post_meta( $subscription->id, '_simplify_customer_id', get_post_meta( $renewal_order->id, '_simplify_customer_id', true ) );
+		$renewal_order_id = version_compare( WC_VERSION, '3.0', '<' ) ? $renewal_order->id : $renewal_order->get_id();
+		update_post_meta( $subscription->id, '_simplify_customer_id', get_post_meta( $renewal_order_id, '_simplify_customer_id', true ) );
```
Comment (gedex) [inline includes/class-wc-addons-gateway-simplify-commerce.php]: afaik, subscription extends `WC_Order`. Could trigger the same notice. Maybe update later once Subscription branch for 3.0.0 is ready.

File: includes/class-wc-addons-gateway-simplify-commerce.php [hunk 3]
```diff
@@ -430,19 +444,21 @@ public function process_pre_order_release_payment( $order ) {
...
+			$customer_id = get_post_meta( $order_id, '_simplify_customer_id', true );
 
 			if ( ! $customer_id ) {
 				return new WP_Error( 'simplify_error', __( 'Customer not found', 'woocommerce' ) );
 			}
 
 			// Charge the customer
 			$payment = Simplify_Payment::createPayment( array(
-				'amount'              => $order->order_total * 100, // In cents.
+				'amount'              => absint( $order->get_total() * 100 ), // In cents.
```
Comment (gedex) [inline includes/class-wc-addons-gateway-simplify-commerce.php]: same feedback in `absint`.

File: includes/class-wc-gateway-simplify-commerce.php [hunk 4]
```diff
@@ -496,7 +496,7 @@ protected function process_standard_payments( $order, $cart_token = '', $custome
 	 * @return bool|WP_Error
 	 */
 	public function do_payment( $order, $amount = 0, $token = array() ) {
-		if ( $amount * 100 < 50 ) {
+		if ( absint( $amount * 100 ) < 50 ) {
```
Comment (gedex) [inline includes/class-wc-gateway-simplify-commerce.php]: same feedback in `absint`

**Comment by bor0** [comment]
@gedex one more round please.

**Comment by gedex** [comment]
This one is also available in wporg so it'll be different deploy process.

**Pull request merged**